### PR TITLE
Adds "option push_timeout" to allow more time for curl.

### PR DESCRIPTION
**Pull request by wylis-hodor**
Also adds log of curl ERRORBUFFER in SSL_CONNECT_ERROR case.

### CODE REVIEW

File: modules/http_push.c [hunk 1]
```diff
@@ -81,6 +85,13 @@ static int nw_http_push_start_acquire_data(struct nodewatcher_module *module,
         free(client_certificate);
         free(client_key);
 
+        /* provide a buffer to store errors in */
+        char errbuf[CURL_ERROR_SIZE];
+        curl_easy_setopt(curl, CURLOPT_ERRORBUFFER, errbuf);
+
+        /* set the error buffer as empty before performing a request */
+        errbuf[0] = 0;
```
Comment (mitar) [inline modules/http_push.c]: I would prefer `'\0'` here, as style.

File: modules/http_push.c [hunk 2]
```diff
@@ -56,7 +56,11 @@ static int nw_http_push_start_acquire_data(struct nodewatcher_module *module,
       CURL *curl = curl_easy_init();
       if (curl) {
         curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, nw_http_push_ignore_data);
-        curl_easy_setopt(curl, CURLOPT_TIMEOUT, 5);
+        int timeout = nw_uci_get_int(uci, "nodewatcher.@agent[0].push_timeout");
+        if (timeout < 5) {
```
Comment (mitar) [inline modules/http_push.c]: Maybe `timeout == 0`.

Comment (wylis-hodor) [reply modules/http_push.c]: Is that what nw_uci_get_int() returns if its not set?

Comment (mitar) [reply modules/http_push.c]: Yes, it seems so.

Comment (wylis-hodor) [reply modules/http_push.c]: OK I see it. I agree "== 0" looks good.  I will make these changes.

File: modules/http_push.c [hunk 3]
```diff
@@ -81,6 +85,13 @@ static int nw_http_push_start_acquire_data(struct nodewatcher_module *module,
         free(client_certificate);
         free(client_key);
 
+        /* provide a buffer to store errors in */
```
Comment (kostko) [inline modules/http_push.c]: Please fix comments so they start with a capital letter and end with a dot (also below).

File: modules/http_push.c (hunk 1 above)
Comment (kostko) [reply modules/http_push.c]: I agree.

File: modules/http_push.c [hunk 4]
```diff
@@ -96,6 +107,14 @@ static int nw_http_push_start_acquire_data(struct nodewatcher_module *module,
           case CURLE_SSL_PINNEDPUBKEYNOTMATCH:
 #endif
           case CURLE_PEER_FAILED_VERIFICATION: push_result = "peer_verify_error"; break;
+          case CURLE_SSL_CONNECT_ERROR:
+            push_result = "ssl_connect_error";
+            if (strlen(errbuf)) {
+              syslog(LOG_WARNING, "buf: %s", errbuf);
```
Comment (kostko) [inline modules/http_push.c]: Can you change the warning to be something more descriptive? E.g., `http-push: SSL connect error: %s`?

File: modules/http_push.c [hunk 5]
```diff
@@ -96,6 +107,14 @@ static int nw_http_push_start_acquire_data(struct nodewatcher_module *module,
...
+            } else {
+              syslog(LOG_WARNING, "err: %s", curl_easy_strerror(result));
```
Comment (kostko) [inline modules/http_push.c]: Can you change the warning to be something more descriptive? E.g., `http-push: SSL connect error: %s`?

File: modules/http_push.c (hunk 2 above)
Comment (kostko) [reply modules/http_push.c]: Yes, this should be changed.

**Review by kostko — CHANGES_REQUESTED** [review]
Looks good, only a few minor comments need to be addressed.

**Comment by wylis-hodor** [comment]
I squashed my commits from this review into the original commit.  I am not sure if you can see the edits now or if i should submit a new pull request and close this one.  I guess i should have just committed the edits and let you squash.

**Comment by mitar** [comment]
Yes, no worries about squashing. You can do it at the end, or we can do it when merging.

**Review by kostko — APPROVED** [review]
Looks good!

**Pull request merged**